### PR TITLE
new mech weapons

### PR DESCRIPTION
**Pull request by Tlaltecuhtli**
:cl:
add: 5 new mech weapons
balance: lowered fire delay for the mech laser
balance: ac2 no longer has 300 ammo now 120
balance: increased energy use of energy weapons
balance: mech rcd no longer takes 100 years to build
balance: mech teleporter uses bluespace crystals instead of diamonds
code: mech materials now have standarized spacing 
/:cl:

**the shitty laser:** now fires faster
**xray gun :** fires xrays 
**incendiary  launcher :** 3x3 clf3 fire, fast cooldown 5 shots before reloading
**acid:** sprays acid and melt your clothes if you dont clean the acid away , inspired by citadel 
**explosive:** it makes a hole in space long detonation timer, good for department demolitons also ur fault for standing still on top of a grenade if u die 
**riot foam:** shoots riot foam , can be used on a ripley too
**e_bola launcher:** shoots bolas

UPDATE: explosive grenade nerfed by half
TODO: harm beam

### CODE REVIEW

**Comment by PKPenguin321** [comment]
you probably want to atomize some of these changes into separate PRs, this looks way too broad in scope

that aside what are the new mech weapons exactly? i wanna know

**Comment by Tlaltecuhtli** [comment]
it seems a lot but its literally 80% adding spaces and revelant changes are in the op

File: code/modules/research/designs/mechfabricator_designs.dm [hunk 1]
```diff
@@ -575,16 +485,16 @@
 	id = "borg_upgrade_rename"
 	build_type = MECHFAB
 	build_path = /obj/item/borg/upgrade/rename
-	materials = list(MAT_METAL=35000)
-	construction_time = 120
+	materials = list(MAT_METAL = 5000)
```
Comment (Tlaltecuhtli) [inline code/modules/research/designs/mechfabricator_designs.dm]: borg upgrades changes here

File: code/modules/research/designs/mecha_designs.dm [hunk 2]
```diff
@@ -253,8 +218,8 @@
 	id = "mech_rcd"
 	build_type = MECHFAB
 	build_path = /obj/item/mecha_parts/mecha_equipment/rcd
-	materials = list(MAT_METAL=30000,MAT_GOLD=20000,MAT_PLASMA=25000,MAT_SILVER=20000)
-	construction_time = 1200
+	materials = list(MAT_METAL = 30000, MAT_GLASS = 20000, MAT_GOLD = 20000, MAT_SILVER = 20000, MAT_PLASMA = 25000)
+	construction_time = 200
```
Comment (Tlaltecuhtli) [inline code/modules/research/designs/mecha_designs.dm]: rcd construction time change, 1200 is obv a typo no one sane would think this is right

**Comment by ExcessiveUseOfCobblestone** [comment]
> here are mech things
>> oh btw massively reduced borg upgrade costs

WHEN WILL HE LEARN

File: code/modules/research/designs/mecha_designs.dm [hunk 3]
```diff
@@ -243,7 +208,7 @@
 	id = "mech_teleporter"
 	build_type = MECHFAB
 	build_path = /obj/item/mecha_parts/mecha_equipment/teleporter
-	materials = list(MAT_METAL=10000,MAT_DIAMOND=10000)
+	materials = list(MAT_METAL = 10000, MAT_GLASS = 4000, MAT_BLUESPACE = 10000)
```
Comment (Tlaltecuhtli) [inline code/modules/research/designs/mecha_designs.dm]: this thing was from before bluespace was a mineral just an update to modernity

**Comment by Kagus** [comment]
From a cursory glance, this appears to add a mech-mounted version of the Xray rifle, a mech sniper rifle with penetrator rounds, and three new types of grenade launcher (incendiary, acid splash and explosive).

Taking into consideration the fact that the barman's shotgun can fire any shotgun round, and that cold temperatures are lethal to slimes, I'd say this is fairly balanced.

---
Oh, by the way, that incendiary grenade isn't going to do diddly unless it somehow manages to directly hit someone with one of the splashes. You either need to have them create reaction upon combining, or pick chemicals that have on-turf reactions.

Or just remove it, because it is silly.

**Comment by Tlaltecuhtli** [comment]
the incendiary works fine i tested it

File: code/modules/research/designs/mechfabricator_designs.dm (hunk 1 above)
Comment (PKPenguin321) [reply code/modules/research/designs/mechfabricator_designs.dm]: yeah but see borg upgrade cost changes shouldnt be in a PR titled "new mech weapons" in the first place

File: code/modules/research/designs/mecha_designs.dm (hunk 2 above)
Comment (PKPenguin321) [reply code/modules/research/designs/mecha_designs.dm]: same for this

**Comment by kevinz000** [comment]
:-1: going to be honest and I'll probably get drowned out by 100 upvotes but holy fuck this is the most overpowered thing I've seen in ages you mech-murderbone main.

**Comment by kevinz000** [comment]
also your new weapons aren't in CL or pr description you shouldn't expect to codedive to find out what you did.

**Comment by Tlaltecuhtli** [comment]
it requires illegal tech and its balanced by having a long cooldown between shots

**Comment by ExcessiveUseOfCobblestone** [comment]
@Tlaltecuhtli I agree I think that's costly, but that's not an excuse to throw it in with mech changes.

**Comment by WJohn** [comment]
fucking remove the wall penetrating sniper rifle *immediately* holy fuck that's AWFUL game design it should not ever be a question

**Comment by PKPenguin321** [comment]
its not shitposting
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#pull-request-process
>If you are proposing multiple changes, which change many different aspects of the code, you are expected to section them off into different pull requests in order to make it easier to review them and to deny/accept the changes that are deemed acceptable.

**Comment by Kagus** [comment]
I have to admit that I'm suffering from an intense desire to shitpost about the mech weapons, including both the mech-mounted xray lasers and penetrator rifles as well as the three flavors of hull breaching and property destruction in grenade form. I am trying to be better about this, however.

In short: Mechs do not even remotely need a buff, these weapons are decidedly a buff.

**Comment by zxaber** [comment]
>balance: mech rcd no longer takes 100 years to build

Yes please

**Comment by Bawhoppen** [comment]
From what I can garner, this seems pretty OP

**Comment by Tlaltecuhtli** [comment]
in which way op? did you test them before commenting

**Comment by Bawhoppen** [comment]
No I did not test them, but it looks like they might be

**Comment by AlexTheSysop42** [comment]
The AC2 is a burst-fire machine gun, it's supposed to hold lots of rounds

**Comment by Tlaltecuhtli** [comment]
by how mech balistic works you use your cell on reload and having 300 rounds means you dont really need to use your power which makes is it able with bs builds like slapfest gygax

File: code/game/mecha/equipment/weapons/weapons.dm [hunk 4]
```diff
@@ -252,6 +262,16 @@
 	projectile_energy_cost = 15
 	harmful = TRUE
 
+/obj/item/mecha_parts/mecha_equipment/weapon/ballistic/sniper
```
[the remark this answers is not in the document]
Comment (MimicFaux) [reply code/game/mecha/equipment/weapons/weapons.dm]: The crew can't make mediguns like nukeops can get, but they can make mech-mounted versions.

[the remark this answers is not in the document]
Comment (iskyp) [reply code/game/mecha/equipment/weapons/weapons.dm]: the station can make rocket pods for mechs but WATCH OUT GUYS WE CAN'T LET THEM HAVE SNIPERS

**Comment by AlexTheSysop42** [comment]
But the whole point of slapfest gygax is the UNLIMITED POWER

File: code/game/mecha/equipment/weapons/weapons.dm (hunk 4 above)
[the remark this answers is not in the document]
Comment (ExcessiveUseOfCobblestone) [reply code/game/mecha/equipment/weapons/weapons.dm]: > syndicate have a monopoly on ballistics but with a scope

this is why NT should always have e-weapons and syndicate ballistic

**Comment by Barhandar** [comment]
The only good part of this PR is the mech RCD buildtime fix.

**Comment by WJohn** [comment]
I am still SUPER strongly against giving mechs a sniper rifle. Sniper rifles are fucking terrible gameplay for the victim as they can't see or hear it coming as it's out of sight, they have guaranteed knockdown that lasts long enough for a follow-up shot, and they also frequently dismember a limb if you miraculously don't just get hit twice in succession so that's miserable.

We ought to remove sniper rifles from nuke ops as well, they were a mistake.

**Comment by Kagus** [comment]
I do actually quite like the idea of a mech-sized Donksoft gun though. Seems like it would be an incredible mess (and potentially at least as deadly as lethal ammo), but the concept is just wonderfully silly.

Today's Moment of Autism: For future reference, it's technically spelled "anti-materi**e**l" rifle, with an **e**. This is because the French hold copyrights on the English language.

File: code/game/mecha/equipment/weapons/weapons.dm [hunk 5]
```diff
@@ -85,16 +85,26 @@
...
 	desc = "A weapon for combat exosuits. Shoots heavy lasers."
 	icon_state = "mecha_laser"
-	energy_drain = 60
+	energy_drain = 80
 	projectile = /obj/item/projectile/beam/laser/heavylaser
 	fire_sound = 'sound/weapons/lasercannonfire.ogg'
 
+/obj/item/mecha_parts/mecha_equipment/weapon/energy/xray
+	equip_cooldown = 10
+	name = "\improper xray laser"
```
Comment (ShizCalev) [inline code/game/mecha/equipment/weapons/weapons.dm]: X-ray

File: code/game/mecha/equipment/weapons/weapons.dm [hunk 6]
```diff
@@ -385,6 +395,73 @@
...
+	icon_state = "mecha_grenadelnchr"
+	projectile = /obj/item/restraints/legcuffs/bola/energy/mech
+	fire_sound = 'sound/weapons/bolathrow.ogg'
+	projectiles = 10
+	missile_speed = 1.5
+	projectile_energy_cost = 250
+	equip_cooldown = 10
+
+/obj/item/mecha_parts/mecha_equipment/weapon/ballistic/launcher/incendiary
+	name = "\improper incendiary grenade launcher"
```
Comment (ShizCalev) [inline code/game/mecha/equipment/weapons/weapons.dm]: no need for this \improper

File: code/game/mecha/equipment/weapons/weapons.dm [hunk 7]
```diff
@@ -385,6 +395,73 @@
...
+	harmful = TRUE
+
+/obj/item/mecha_parts/mecha_equipment/weapon/ballistic/launcher/incendiary/proj_init(var/obj/item/grenade/chem_grenade/pyro/mechi/F)
+	var/turf/T = get_turf(src)
+	message_admins("[ADMIN_LOOKUPFLW(chassis.occupant)] fired a [src] in [ADMIN_VERBOSEJMP(T)]")
+	log_game("[key_name(chassis.occupant)] fired a [src] in [AREACOORD(T)]")
+	addtimer(CALLBACK(F, /obj/item/grenade/chem_grenade/pyro/mechi.proc/prime), det_time)
+
+/obj/item/mecha_parts/mecha_equipment/weapon/ballistic/launcher/acid
+	name = "\improper acid grenade launcher"
```
Comment (ShizCalev) [inline code/game/mecha/equipment/weapons/weapons.dm]: no need for this \improper

File: code/game/mecha/equipment/weapons/weapons.dm [hunk 8]
```diff
@@ -385,6 +395,73 @@
...
+	harmful = TRUE
+
+/obj/item/mecha_parts/mecha_equipment/weapon/ballistic/launcher/acid/proj_init(var/obj/item/grenade/chem_grenade/mecha/F)
+	var/turf/T = get_turf(src)
+	message_admins("[ADMIN_LOOKUPFLW(chassis.occupant)] fired a [src] in [ADMIN_VERBOSEJMP(T)]")
+	log_game("[key_name(chassis.occupant)] fired a [src] in [AREACOORD(T)]")
+	addtimer(CALLBACK(F, /obj/item/grenade/chem_grenade/mecha.proc/prime), det_time)
+
+/obj/item/mecha_parts/mecha_equipment/weapon/ballistic/launcher/explosive
+	name = "\improper grenade launcher"
```
Comment (ShizCalev) [inline code/game/mecha/equipment/weapons/weapons.dm]: no need for this \improper

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Pull request closed without merging**